### PR TITLE
Modularization/fix glossary associations

### PR DESCRIPTION
**Pull request by noah-paige**
### Feature or Bugfix

- Bugfix

### Detail
- Unable to load Glossary Term Associations for `DataStorageLocation`
  - Pass the correct target_type of `Folder` when setting Glossary Term Links

- Unable to load Glossary Term Associations when either Dashboards or Datasets Module Disabled
  - We have inline fragments on the `getGlossary` graphQL API that will throw an error if the ObjectType is not defined (i.e. when Dashboards or Datasets are disabled) 
  - Since we only require a common field returned by any of the Term Association Objects (we only need `label` field) we can remove the inline fragments so there is no need to have object types defined when loading the glossary term associations
  - Also cleaned up some of the other API Get calls on glossary that return additional information that is unused

**NOTE: This PR is branched off of #731 and should be merged ONLY AFTER the prior is to `main-v2`**

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

NA 
```
- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by noah-paige** [comment]
Local Testing:

- [ ] Create, Get, Delete Glossary
- [ ] Create, Get, Delete Category
- [ ] Create, Get, Delete Term
- [ ] Add Glossary Term to Folder and Approve, Reject
- [ ] Add Glossary Term to Dataset and Approve, Reject
- [ ] Search By Glossary Term Filter on Catalog
- [ ] Disable Dashboards Only - Glossary Still Works
- [ ] Disable Datasets Only  - Glossary Still Works
- [ ] Disable both Datasets and Dashboards - Glossary Not Available

AWS Testing:

- [ ] Create, Get, Delete Glossary
- [ ] Create, Get, Delete Category
- [ ] Create, Get, Delete Term
- [ ] Add Glossary Term to Folder and Approve, Reject
- [ ] Add Glossary Term to Dataset and Approve, Reject
- [ ] Add Glossary Term to DatasetTable and Approve, Reject
- [ ] Add Glossary Term to Dashboard and Approve, Reject
- [ ] Search By Glossary Term Filter on Catalog

**Comment by dlpzx** [comment]
@noah-paige thanks! I am doing a deploying it in AWS just in case, but overall it looks good!

**Review by dlpzx — APPROVED** [review]
Looks good

File: backend/dataall/modules/catalog/api/types.py [hunk 1]
```diff
@@ -250,10 +249,11 @@
     ],
 )
 
-GlossaryTermLinkTarget = gql.Union(
+GlossaryTermLinkTarget = gql.ObjectType(
     name='GlossaryTermLinkTarget',
-    type_registry=GlossaryRegistry,
-    resolver=target_union_resolver,
```
Comment (dbalintx) [inline backend/dataall/modules/catalog/api/types.py:256]: `target_union_resolver`  import could be removed, the resolver method itself as well + maybe `GlossaryRegistry.find_object_type` as well

Comment (noah-paige) [reply backend/dataall/modules/catalog/api/types.py:256]: Thanks @dbalintx - will take this up as part of clean up task in the backlog #741

**Pull request merged**